lib/src/shared/main/java/com/couchbase/lite/AbstractReplicator.java

```diff
@@ -390,6 +391,60 @@ public void stop() {
     @NonNull
     public Status getStatus() { return status.copy(); }
 
+    @NonNull
+    public Set<String> getPendingDocumentIds() throws CouchbaseLiteException {
+        if (config.getReplicatorType().equals(ReplicatorConfiguration.ReplicatorType.PULL)) {
+            throw new CouchbaseLiteException(
+                "Pending Document IDs are not supported on pull-only replicators.",
+                Log.LOGGING_DOMAINS_TO_C4.get(DOMAIN),
+                CBLError.Code.UNSUPPORTED);
+        }
+
+        final C4Replicator repl;
+        synchronized (lock) { repl = c4repl; }
+
+        if (repl == null) {
+            Log.i(DOMAIN, "%s: Call to getPendingDocumentIds with unstarted replicator", this);
+            return Collections.emptySet();
+        }
+
+        final Set<String> pending;
+        try { pending = c4repl.getPendingDocIDs(); }
+        catch (LiteCoreException e) {
+            Log.w(DOMAIN, "Error while fetching pending documentIds: %d/%d", e.domain, e.code);
+            return Collections.emptySet();
```

Comment:
Should we just throw CouchbaseLiteException here?

Reply:
I would prefer that.  I was going for parity with iOS, which does exactly this.

Reply:
Done